### PR TITLE
Add support for Python style check with Black

### LINKED ISSUE #487: Support Python formatting with Black
Hi yall and thank you for making this wonderful tool. I just recently discovered it (after having previously overlooked it because I was confused by the name). I'm testing it out now.

One thing I like to run in my pipelines is [Black](https://black.readthedocs.io/en/stable/) for Python code formatting. I see you have clang-format already supported. Are you open to supporting Black as well? I could probably attempt a PR.

### PR DESCRIPTION
**Pull request by mortenfyhn**
Fix #487.

This seems to work. I've made a test repo where I run this branch of my fork of Industrial CI: https://github.com/mortenfyhn/rospy_test (Using Semaphore CI, see .semaphore/semaphore.yml)

Here is the output of the passing master branch build: https://mortenfyhn.semaphoreci.com/workflows/b69d6f0f-afca-421f-8d22-3aa1b2d829f1

Here is the output of a build where I deliberately mis-formatted a Python file: https://mortenfyhn.semaphoreci.com/workflows/ef3c8bda-8838-4420-99ff-adc1464f4ef1

I mostly just copied and modified `clang_format_check.sh`, and removed as much as I could from it. I may have overlooked something, so please have a look.

### CODE REVIEW

**Comment by mortenfyhn** [comment]
Hi everyone, any updates on this? I just rebased in onto the latest master.

**Comment by ipa-mdl** [comment]
> Hi everyone, any updates on this? I just rebased in onto the latest master.

Sry,  I did not have much time for this lately.
I would like to review all pending PRs in September.

**Comment by mortenfyhn** [comment]
> > Hi everyone, any updates on this? I just rebased in onto the latest master.
> 
> Sry, I did not have much time for this lately.
> I would like to review all pending PRs in September.

I understand, thanks for the feedback, that sounds good!

File: industrial_ci/src/tests/black_check.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+function run_black_check() {
```
Comment (ipa-mdl) [inline industrial_ci/src/tests/black_check.sh:3]: This needs to be  updated after #558 and to use the new pip function, but I will do so in the next days.

Comment (mortenfyhn) [reply industrial_ci/src/tests/black_check.sh:3]: That sounds good! Let me know if you want me to fix anything.

File: industrial_ci/src/tests/black_check.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  DOCKER_IMAGE="$DOCKER_BASE_IMAGE" ici_require_run_in_docker # this script must be run in docker
+
+  ici_time_start install_black
+  ici_quiet ici_asroot apt-get install -qq -y python3-pip
+  ici_quiet ici_asroot pip3 -q install black
```
Comment (ipa-mdl) [inline industrial_ci/src/tests/black_check.sh:8]: Is there any issue with running the python3 version of black with python2 code?

Comment (mortenfyhn) [reply industrial_ci/src/tests/black_check.sh:8]: Should be fine: https://github.com/psf/black#installation

Comment (mortenfyhn) [reply industrial_ci/src/tests/black_check.sh:8]: I assume it just checks the python shebang.

**Comment by ipa-mdl** [comment]
@mortenfyhn: I have rebase your branch and updated the install a little bit: https://github.com/ipa-mdl/industrial_ci/tree/feature/black
Please review this and push it onto your branch :)

Unfortunately, `ROS_DISTRO` has to be specified for now.

**Comment by mortenfyhn** [comment]
I had a look at your changes, and it seems good! Tested locally to make sure that it works (`rosrun industrial_ci run_ci ROS_DISTRO=noetic ROS_REPO=main BLACK_CHECK=true`) on a package with Python code failing and passing black.

**Comment by mortenfyhn** [comment]
...and rebased!

**Review by ipa-mdl — APPROVED** [review]

**Comment by mortenfyhn** [comment]
Awesome, thanks!

**Pull request merged**